test/e2e/k8s_util.go

```diff
@@ -775,18 +763,25 @@ func (data *TestData) GetTier(name string) (*crdv1beta1.Tier, error) {
 // UpdateTier is a convenience function for updating an Antrea Policy Tier.
 func (data *TestData) UpdateTier(tier *crdv1beta1.Tier) (*crdv1beta1.Tier, error) {
 	log.Infof("Updating tier %s", tier.Name)
-	updatedTier, err := data.crdClient.CrdV1beta1().Tiers().Update(context.TODO(), tier, metav1.UpdateOptions{})
-	return updatedTier, err
+	return data.crdClient.CrdV1beta1().Tiers().Update(context.TODO(), tier, metav1.UpdateOptions{})
+}
+
+func isReferencedError(err error) bool {
+	if status, ok := err.(apierrors.APIStatus); ok || errors.As(err, &status) {
+		// The message is set by deleteValidate of tierValidator when deleting a Tier that is referenced by any policies.
+		return strings.Contains(status.Status().Message, "is referenced by")
+	}
+	return false
 }
 
 // DeleteTier is a convenience function for deleting an Antrea Policy Tier with specific name.
+// To avoid flakes caused by antrea-controller not in sync with kube-apiserver, it retries a few times if the failure is
+// because the Tier is still referenced.
 func (data *TestData) DeleteTier(name string) error {
-	_, err := data.crdClient.CrdV1beta1().Tiers().Get(context.TODO(), name, metav1.GetOptions{})
-	if err != nil {
-		return fmt.Errorf("unable to get tier %s: %w", name, err)
-	}
 	log.Infof("Deleting tier %s", name)
-	if err = data.crdClient.CrdV1beta1().Tiers().Delete(context.TODO(), name, metav1.DeleteOptions{}); err != nil {
+	if err := retry.OnError(retry.DefaultRetry, isReferencedError, func() error {
```

Comment:
does this have a reasonable timeout?

Reply:
yes, a exponential backoff with 5 retries with initial interval of 10ms

Reply:
Here I use `DefaultRetry` which doesn't increase interval exponentially, it retries 5 times with a constant interval (10ms) . I just assume extra 50ms is enough given the flake is not that frequent. If it's proved not enough, we can change it to `DefaultBackoff` which increases interval exponentially.